### PR TITLE
Fix QPDFOutlineDocumentHelper::resolveNamedDest  (fixes #1238)

### LINKED ISSUE #1238: qpdf fails to find existing bookmark targets for specific PDF
Tried with latest release and even latest nightly as of today.
With the attached PDF, the bookmark example program `pdf-bookmarks` and also my own code using the library fail to retrieve target pages for all bookmarks, although they are stored in the PDF and displayed correctly in a PDF reader based on xpdf.

I used the command line
```
pdf-bookmarks --show-targets --lines
```
For other PDF files it worked flawlessly.
Using Haiku nightly here as OS but should be the same in Linux, will try later on request.

So for the attached PDF, this gives me correct bookmarks but not the pages, which I need, too:
```
+-+ [ -> unknown ] Conclusions
| |
| +-+ [ -> unknown ] Confirmation of Research Hypotheses
| |
| +-+ [ -> unknown ] Limitations and Future Work
|   |
|   +-+ [ -> unknown ] Extension to Other Tasks
|   |
|   +-+ [ -> unknown ] Memory Requirements
|   |
|   +-+ [ -> unknown ] Edge Engineering
|   |
|   +-+ [ -> unknown ] Walk-based Inference
|
+-+ [ -> unknown ] Hyper-parameter Settings
| |
| +-+ [ -> unknown ] Chapter 4
| |
| +-+ [ -> unknown ] Chapter 5
| |
| +-+ [ -> unknown ] Chapter 6
|
+-+ [ -> unknown ] Bibliography
```
The test PDF is used is:
[Textual Relation Extraction With Edge-Oriented Graph Neural Models.pdf](https://github.com/user-attachments/files/16197330/Textual.Relation.Extraction.With.Edge-Oriented.Graph.Neural.Models.pdf)

### PR DESCRIPTION
**Pull request by m-holger**
Handle named destinations where the entry is a dictionary with /D entry.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,8 @@
+2024-07-13  M Holger  <m.holger@qpdf.org>
+
+	* Bug fix: handle named destinations where the entry is a 
+	dictionary with /D entry instead of an explicit destination.
```
Comment (jberkenbilt) [inline ChangeLog:4]: Please add `Fixes #1238`

**Review by jberkenbilt — APPROVED** [review]
Thanks for fixing this, adding a test, and cleaning up the code in a separate commit.

**Comment by jberkenbilt** [comment]
This is a fix for #1238

File: ChangeLog (hunk 1 above)
Comment (m-holger) [reply ChangeLog:4]: Done, including commit message.

Comment (m-holger) [reply ChangeLog:4]: Also fixed the line endings.

**Pull request merged**